fairscale/experimental/wgit/repo.py

```diff
@@ -170,16 +177,19 @@ def path(self) -> Path:
             self._exists(self.wgit_parent)
         return self._repo_path
 
-    def _get_tracked_metadata_files(self) -> Dict:
-        """Goes through the weigit directory to find the metadata file of the files being tracked."""
+    def _get_metdata_files(self) -> Dict:
+        """Walk the directories that contain the metadata files and check the status of those files,
+        whether they have been modified or not.
+        """
         metadata_d = dict()
-        for file in self.path.iterdir():
-            if file.is_file() and self._is_metadata_file(file):
-                try:
-                    metadata_d[file.name] = self._is_file_modified(file)
-                except json.JSONDecodeError as error:
-                    # files that are not json files are simply skipped
-                    pass
+        for file in self.path.iterdir():  # iterate over the .wgit directory
```

Comment:
Does this loop actually walk the dir tree or it just iterate over the first level children of the dir? Maybe I missed how you walk the dir tree here? Maybe os.walk() function should be used here?

Reply:
This loop just iterates over the first level children of `.wgit` and ignores the `.wgit` internal directories and files. We do the directory walk (to deeper level in the non-ignored directories) using `for path in file.rglob("*"):` in the line 189. Since, I used `pathlib` throughout the implementation, I decided to use `pathlib` functionalities for dir walk for consistency.